calyx-py/test/correctness/tuple.py

```diff
@@ -0,0 +1,115 @@
+# pylint: disable=import-error
+import calyx.builder as cb
+
+
+def insert_tuplify(prog, name, w1, w2):
+    """Inserts the combinational component `tuplify` into the program.
+
+    It takes two inputs, `fst` (width `w1`) and `snd` (width `w2`),
+    and outputs a tuple (width `w1 + w2`) that contains `fst` and `snd`.
+    """
+
+    width = w1 + w2
+
+    comp = prog.comb_component(name)
```

Comment:
Not that it really matters at all, but is there a reason why you chose not to do `std_cat` for this? (And `std_bit_slice`) for the "untuplify"?

Reply:
I think I did use `std_bit_slice`! Did I not, somehow? Also I don't know what `std_cat` is, and I didn't see it in the docs. If you tell me more about it or point me to a resource, I'll use it!

Reply:
Yeah you're completely right, that's my bad (I didn't read your `untplify` implementation). 

`std_cat` basically takes in two signals and concatenates them together. It's implementation is here: https://github.com/calyxir/calyx/blob/a27bb66c2c07a9063741bc0e0aabd5087285260e/primitives/core.sv#L53.

Reply:
Super, thanks! Used `cat` after adding support in the builder and the ast!